parametric/conftest.py

```diff
@@ -316,13 +316,15 @@ def wait_for_num_traces(self, num: int, clear: bool = False) -> List[Trace]:
         num_received = None
         for i in range(20):
             try:
-                traces = self.traces(clear=clear)
+                traces = self.traces(clear=False)
             except requests.exceptions.RequestException:
                 pass
             else:
                 num_received = len(traces)
                 if num_received == num:
-                    return traces
+                    if clear:
+                        self.clear()
```

Comment:
Clean up once the number of expected traces received.

Reply:
We could but since we raise an exception when we haven't received expected number of traces it won't make any difference.